Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py

```diff
@@ -654,24 +663,21 @@ def C2Fw(self, sname):
             height_data, height_error = np.asarray(height_data), np.asarray(height_error)
 
             # calculate EISF and EISF error
-            total = height_data + amplitude_data
-            EISF_data = height_data / total
-            total_error = height_error**2 + amplitude_error**2
-            EISF_error = EISF_data * np.sqrt((height_error**2 / height_data**2) + (total_error / total**2))
+            eisf_data, eisf_error = _calculate_eisf(height_data, height_error, amplitude_data, amplitude_error)
 
             # interlace amplitudes and widths of the peaks
             y.append(np.asarray(height_data))
-            for amp, width, EISF in zip(amplitude_data, width_data, EISF_data):
+            for amp, width, eisf in zip(amplitude_data, width_data, eisf_data):
```

Comment:
This loop seems a bit odd but I think an equivalent way to do it would be to stack the arrays and then flatten rather than loop over each element?
`y = np.vstack((amplitude_data, width_data, eisf_data)).flatten("F")`
Not sure if that's faster - probably less readable though!